### PR TITLE
(Splink 4) Comparisons for remaining distance functions

### PR DESCRIPTION
**Pull request by RobinL**
Add:
`DamerauLevenshteinAtThresholds`
`JaccardAtThresholds`
`JaroAtThresholds`
`JaroWinklerAtThresholds`

<details>
<summary>
Test script
</summary>

```python
from splink.datasets import splink_datasets
from splink.duckdb.blocking_rule_library import block_on
from splink.comparison_library import (
    DamerauLevenshteinAtThresholds,
    JaccardAtThresholds,
    JaroAtThresholds,
    JaroWinklerAtThresholds,
    ExactMatch
)

from splink.duckdb.linker import DuckDBLinker

JaroWinklerAtThresholds("first_name", 0.7).create_comparison_dict("spark")
df = splink_datasets.fake_1000

settings = {
    "probability_two_random_records_match": 0.01,
    "link_type": "dedupe_only",
    "blocking_rules_to_generate_predictions": [
        block_on("first_name"),
        block_on("surname"),
    ],
    "comparisons": [
        DamerauLevenshteinAtThresholds("first_name", 2),
        JaccardAtThresholds("surname", [0.95, 0.9]),
        JaroAtThresholds("dob"),
        # ExactMatch("city", term_frequency_adjustments=True),
        JaroWinklerAtThresholds("city"),
        ExactMatch("email"),
    ],
    "retain_intermediate_calculation_columns": True,
    "additional_columns_to_retain": ["cluster"],
    "max_iterations": 10,
    "em_convergence": 0.01,
}


linker = DuckDBLinker(df, settings)

linker.estimate_u_using_random_sampling(target_rows=1e5)
display(linker.match_weights_chart())
linker.save_model_to_json()
```

</details>

### CODE REVIEW

**Review by ADBond — APPROVED** [review]
LGTM 👍

**Comment by RobinL** [comment]
Going to hold off merging this until the input expression PR is merged, since this will then need some minor updates

**Pull request merged**
